torch/_export/non_strict_utils.py

```diff
@@ -287,6 +287,61 @@ def produce_guards_and_solve_constraints(
         raise constraint_violation_error
 
 
+def replace_non_strict_symbol_sources(shape_env, arg_names):
+    """
+    For non-strict mode, we map shape symbols to sources that don't match strict,
+    e.g. L["args"][0][..] for args; L["args"][1][..] for kwargs.
+    For prettifying runtime asserts, we'd like to map these back to the original names
+    in the forward() signature. These sources remain in the ShapeEnv, even when they're used
+    for later stages that are independent of strict/non-strict mode (e.g. run_decompositions()),
+    so here we just permanently replace the sources with what they'd look like for strict.
+    """
+    from dataclasses import asdict
+
+    if shape_env.dim_constraints is None:
+        return  # no dynamic shapes
+
+    def _replace_sources(source):
```

Comment:
cc @angelayi who made a good point about not doing this because runtime asserts are on a graph module whose placeholder names are divorced from source already, so it might be better to just use the placeholder names here.